frontend/src/components/WeatherCard.js

```diff
@@ -0,0 +1,79 @@
+import useWeather from "../hooks/useWeather";
+import styled from "styled-components/macro";
+import { ChronoUnit, LocalDateTime } from "@js-joda/core";
+
+export default function WeatherCard({ latitude, longitude, dateTime }) {
+  const weather = useWeather(latitude, longitude);
+
+  function findWeatherAtEventTime(weather, dateTime) {
+    if (!weather) {
+      return undefined;
+    }
+    if (
+      LocalDateTime.parse(dateTime).compareTo(LocalDateTime.now().plusDays(4)) >
+      0
+    ) {
+      return undefined;
+    } else {
+      return weather.list.find(
+        (item) =>
+          LocalDateTime.parse(item.dt_txt.replace(" ", "T")).until(
+            LocalDateTime.parse(dateTime),
+            ChronoUnit.HOURS
+          ) < 3
+      );
+    }
+  }
```

Comment:
Consider making this section more readable.

e.g. something like

```suggestion
    if (!weather || !eventDateIsWithinNextFourDays()) {
      return undefined;
    } else {
     parseLocalTime(); // maybe find a better name here 
  }
```